Libraries/LibHTML/CSS/StyleProperties.h

```diff
@@ -10,6 +10,13 @@ class Color;
 class StyleProperties : public RefCounted<StyleProperties> {
 public:
     static NonnullRefPtr<StyleProperties> create() { return adopt(*new StyleProperties); }
+    static NonnullRefPtr<StyleProperties> create(const StyleProperties& properties) {
+        auto style_properties = new StyleProperties();
```

Comment:
Let's just `adopt` this right away to stick to a "no naked `new`" policy.
```suggestion
        auto style_properties = adopt(*new StyleProperties(*this));
```
You may then need to add a copy constructor that copies `m_properties` and `m_font`.
I think `HashMap` supports copy construction so it should be easy.